File: Semestrul 4/MPP/lab04/cs/Service/ServiceImpl.cs
using cs.Domain;
using cs.Repository.Interfaces;

namespace cs.Service;

public class ServiceImpl : IService
{
    private readonly IEventRepository _eventRepository;
    private readonly IParticipantRepository _participantRepository;
    private readonly IRefereeRepository _refereeRepository;
    private readonly IResultRepository _resultRepository;
    private readonly IRefereeDisciplineRepository _refereeDisciplineRepository;

    public ServiceImpl(
        IEventRepository eventRepository,
        IParticipantRepository participantRepository,
        IRefereeRepository refereeRepository,
        IResultRepository resultRepository,
        IRefereeDisciplineRepository refereeDisciplineRepository)
    {
        _eventRepository = eventRepository;
        _participantRepository = participantRepository;
        _refereeRepository = refereeRepository;
        _resultRepository = resultRepository;
        _refereeDisciplineRepository = refereeDisciplineRepository;
    }

    public Referee Login(string username, string password)
    {
        Referee referee = _refereeRepository.FindByUsername(username);
        if (referee != null)
        {
            if (referee.GetPassword().Equals(password))
            {
                return referee;
            }
        }
        return null;
    }

    public List<Event> GetAllEvents()
    {
        return _eventRepository.FindAll().ToList();
    }

    public List<Result> GetResultsByEvent(long eventId)
    {
        return _resultRepository.FindByEvent((int)eventId);
    }

    public void UpdateResult(Result result)
    {
        _resultRepository.Update(result);
    }

    public List<RefereeDiscipline> GetRefereeDisciplines(long refereeId)
    {
        return _refereeDisciplineRepository.FindByRefereeId(refereeId);
    }
}

File: Semestrul 4/SDI/lab05/java/src/main/java/mpp/service/ServiceImpl.java
package mpp.service;

import mpp.domain.*;
import mpp.repository.interfaces.*;

import java.util.List;
import java.util.Optional;

public class ServiceImpl implements Service {
    private final EventRepository eventRepository;
    private final ParticipantRepository participantRepository;
    private final RefereeRepository refereeRepository;
    private final ResultRepository resultRepository;
    private final RefereeDisciplineRepository refereeDisciplineRepository;

    public ServiceImpl(EventRepository eventRepository, ParticipantRepository participantRepository,
                       RefereeRepository refereeRepository, ResultRepository resultRepository, RefereeDisciplineRepository refereeDisciplineRepository) {
        this.eventRepository = eventRepository;
        this.participantRepository = participantRepository;
        this.refereeRepository = refereeRepository;
        this.resultRepository = resultRepository;
        this.refereeDisciplineRepository = refereeDisciplineRepository;
    }

    @Override
    public Referee login(String username, String password) {
        Optional<Referee> refereeOpt = refereeRepository.findByUsername(username);
        if (refereeOpt.isPresent()) {
            Referee referee = refereeOpt.get();
            if (referee.getPassword().equals(password)) {
                return referee;
            }
        }
        return null;
    }

    @Override
    public List<Event> getAllEvents() {
        return (List<Event>) eventRepository.findAll();
    }
    @Override
    public List<Result> getResultsByEvent(Long eventId) {
        return resultRepository.findByEvent(eventId.intValue());
    }

    @Override
    public void updateResult(Result result) {
        resultRepository.update(result);
    }
    @Override
    public List<RefereeDiscipline> getRefereeDisciplines(Long refereeId) {
        return refereeDisciplineRepository.findByRefereeId(refereeId);
    }
}